packages/card-support/src/components/DownloadClientData/index.tsx

```diff
@@ -93,40 +93,45 @@ const DownloadClientData: React.FC<DownloadClientDataProps> = (props: DownloadCl
     return current > moment().startOf('day');
   };
 
-  React.useEffect(() => {
-    const serve = new opensrpServiceClass(accessToken, opensrpBaseURL, OPENSRP_URL_USER_ASSIGNMENT);
-    serve
-      .list()
-      .then((assignment: UserAssignment) => {
-        const { jurisdictions } = assignment;
-        const defaultLocationId = jurisdictions[0];
-        setDefaultLocationId(defaultLocationId);
-        const serve = new opensrpServiceClass(
-          accessToken,
-          opensrpBaseURL,
-          OPENSRP_URL_LOCATION_HIERARCHY
-        );
-        serve
-          // eslint-disable-next-line @typescript-eslint/camelcase
-          .read(defaultLocationId, { is_jurisdiction: true })
-          .then((res: RawOpenSRPHierarchy) => {
-            const hierarchy = generateJurisdictionTree(res);
-            dispatch(fetchAllHierarchiesActionCreator([hierarchy.model]));
-          })
-          .catch((_: Error) => {
-            sendErrorNotification(lang.ERROR_OCCURRED);
-          });
-      })
-      .catch((_: Error) => {
-        sendErrorNotification(lang.ERROR_OCCURRED);
-      });
-  }, [
-    accessToken,
-    opensrpBaseURL,
-    fetchAllHierarchiesActionCreator,
-    opensrpServiceClass,
-    dispatch,
-  ]);
+  interface DefaultLocation {
+    display: string;
+    name: string;
+    uuid: string;
+  }
+
+  // remove '/rest' from base opensrp url (https://some.open.opensrp.url/opensrp/rest/)
+  const BASE_URL = opensrpBaseURL.replace('/rest', '');
+
+  // fetch logged in user data including team assigned to and location assigned to the team
+  const userLocSettings = useQuery(
+    SECURITY_AUTHENTICATE,
+    () => new opensrpServiceClass(accessToken, BASE_URL, SECURITY_AUTHENTICATE).list(),
+    {
+      onError: () => sendErrorNotification(lang.USER_NOT_ASSIGNED_AND_USERS_TEAM_NOT_ASSIGNED),
+      select: (res: { team: { team: { location: DefaultLocation } } }) => res.team.team.location,
```

Comment:
define an interface for this to allow reusability

Reply:
@KipSigei for the whole response object? there's a subset of the interface here https://github.com/opensrp/web/blob/ae106caa0d2de021f6e4c9d333ec2f3b98adabc4/packages/card-support/src/components/DownloadClientData/index.tsx#L96

Reply:
We need a single point of reference for all API response interfaces, probably even a package